lib/OpenQA/Task/Needle/Save.pm

```diff
@@ -101,10 +106,14 @@ sub _save_needle {
 
     # ensure needle dir is up-to-date
     my $git = OpenQA::Git->new({app => $app, dir => $needledir, user => $user});
+    # avoid retrying if cleanup via Git is not configured (then an admin might need to
+    # manually cleanup the repo first)
+    $signal_guard->abort(1) if !$git->enabled || $git->config->{do_cleanup} eq 'no';
     if ($git->enabled) {
         my $error = $git->set_to_latest_master;
         if ($error) {
-            $app->log->error($error);
+            $app->log->error(
+                "Dirty repo detected with do_cleanup disabled. A manual cleanup is required. Error: $error");
```

Comment:
Is it known at this point that `do_cleanup` is disabled?

Reply:
Since in the test that is covering this block `do_cleanup` is set to yes, I think we can confirm that do_cleanup is not disabled.

Reply:
I've changed the error message.